src/services/messagesService.ts

```diff
@@ -208,39 +218,56 @@ export default class MessagesService {
         )
         // Enrich the message with legal_message retrieved from pec-server
         .chain(message =>
-          new TE.TaskEither(
-            new T.Task(() =>
-              // FIXME: add JWT bearer implmenetation
-              this.pecClient.getClient("dummy-bearer-token").getMessage({
-                id: message.content.legal_data?.message_unique_id || "ERROR" // Workaround: legal_data esistence has already be checked
-              })
+          new TokenService()
+            .getPecServerToken(
+              message.fiscal_code,
+              PEC_SERVER_TOKEN_SECRET,
+              PEC_SERVER_TOKEN_EXPIRATION,
+              PEC_SERVER_TOKEN_ISSUER
             )
-              .map(response =>
-                response.mapLeft(_ =>
-                  ResponseErrorInternal(
-                    "Error decoding pecServer getMessage response"
-                  )
-                )
+            .mapLeft(e =>
+              ResponseErrorInternal(
+                `Error computing the PEC Server JWT: ${e.message}`
               )
-              .map(response =>
-                response.map(
-                  E.fromPredicate(isPecServerGetMessageSuccess, e =>
-                    ResponseErrorInternal(
-                      `Error getting the message from pecServer getMessage endpoint (received a ${e.status})` // TODO: disjoint the errors
+            )
+            .chain(pecServerJwt =>
+              new TE.TaskEither(
+                new T.Task(() =>
```

Comment:
```suggestion
              new TE.fromTask
                new T.Task(() =>
```

Reply:
why don't just pass through the error (using a `TaskEither` instead of a `Task`) instead of throwing here?

Reply:
you are right, we can also avoid all the direct constructor call using TE.tryCatch (thanks @AleDore)